tensorflow/core/framework/memory_types.cc

```diff
@@ -64,7 +66,9 @@ void MemoryTypesHelper(const NameRangeMap& name_map,
 
 bool IsFunctionCallOp(const string& op_type) {
```

Comment:
Rather than a list of ops, can we do something like:

```
bool IsFunctionCallOp(const NodeDef& node) {
  for (const auto& attr : node.attr()) {
    if (attr.second.value_case() == AttrValue::kFunc){
      return true;
    }
  }
  return false;
}
```

and then pass a NodeDef here? That should cover everything in the list now.

Reply:
Possibly. A quick test found a few ops which pass the this test but aren't on the list (MapDataset, FlatMapDataset) but those may need to be added to the list anyway.